src/extensions/life_time_extension.rs

```diff
@@ -20,29 +20,29 @@
 //! ```
 use super::{Extension, ExtensionError, ExtensionStruct, ExtensionType, LifetimeExtensionError};
 use crate::codec::{Codec, Cursor};
+use crate::config::Config;
 
 use std::time::{SystemTime, UNIX_EPOCH};
 
 /// The lifetime extension holds a not before and a not after time measured in
 /// seconds since the Unix epoch (1970-01-01T00:00:00Z).
-#[derive(PartialEq, Clone, Debug, Default)]
+#[derive(PartialEq, Clone, Debug)]
 pub struct LifetimeExtension {
     not_before: u64,
     not_after: u64,
 }
 
 impl LifetimeExtension {
-    /// Create a new lifetime extensions with lifetime `t`.
+    /// Create a new lifetime extensions with lifetime `t` (in seconds).
     /// Note that the lifetime is extended 1h into the past to adapt to skewed
     /// clocks.
     pub fn new(t: u64) -> Self {
-        // TODO: #85 make the margin configurable.
-        const LIFETIME_MARGIN: u64 = 60 * 60;
+        let lifetime_margin: u64 = Config::key_package_lifetime_margin().clone();
```

Comment:
This should go away when you change the API and drop the clone. `u64` is copyable.